QuotableKeywordPair can have more than 2 children

# Exception


## Stacktrace

```
java.lang.AssertionError: Assertion failed
	at org.elixir_lang.psi.impl.QuotableKeywordPairImpl.getKeywordValue(QuotableKeywordPairImpl.kt:36)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.getKeywordValue(ElixirPsiImplUtil.java:1017)
	at org.elixir_lang.psi.impl.ElixirKeywordPairImpl.getKeywordValue(ElixirKeywordPairImpl.java:48)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:524)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:513)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:117)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:488)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:346)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:106)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:488)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:346)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:106)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:488)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:346)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:106)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:444)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:480)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.java:94)
	at org.elixir_lang.psi.impl.ProcessDeclarationsImpl.processDeclarationsInPreviousSibling(ProcessDeclarationsImpl.kt:289)
	at org.elixir_lang.psi.impl.ProcessDeclarationsImpl.processDeclarations(ProcessDeclarationsImpl.kt:144)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.processDeclarations(ElixirPsiImplUtil.java:591)
	at org.elixir_lang.psi.impl.ElixirStabBodyImpl.processDeclarations(ElixirStabBodyImpl.java:63)
	at com.intellij.psi.util.PsiTreeUtil.treeWalkUp(PsiTreeUtil.java:1152)
	at org.elixir_lang.psi.scope.variable.MultiResolve.resolveResultList(MultiResolve.java:90)
	at org.elixir_lang.psi.scope.variable.MultiResolve.resolveResultList(MultiResolve.java:68)
	at org.elixir_lang.reference.resolver.Callable.resolveElement(Callable.kt:41)
	at org.elixir_lang.reference.resolver.Callable.resolve(Callable.kt:22)
	at org.elixir_lang.reference.resolver.Callable.resolve(Callable.kt:14)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:149)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:179)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:171)
	at org.elixir_lang.reference.ResolveWithCachingComputable.compute(Callable.kt:593)
	at org.elixir_lang.reference.ResolveWithCachingComputable.compute(Callable.kt:585)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:918)
	at org.elixir_lang.reference.CallableKt.resolveWithCaching(Callable.kt:583)
	at org.elixir_lang.reference.CallableKt.access$resolveWithCaching(Callable.kt:1)
	at org.elixir_lang.reference.Callable.multiResolve(Callable.kt:106)
	at org.elixir_lang.annotator.Callable$annotate$1.visitPlainCall(Callable.kt:98)
	at org.elixir_lang.annotator.Callable$annotate$1.visitNonModuleAttributeCall(Callable.kt:87)
	at org.elixir_lang.annotator.Callable$annotate$1.visitCall(Callable.kt:47)
	at org.elixir_lang.annotator.Callable$annotate$1.visitElement(Callable.kt:78)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:274)
	at org.elixir_lang.psi.impl.ElixirMatchedUnqualifiedNoArgumentsCallImpl.accept(ElixirMatchedUnqualifiedNoArgumentsCallImpl.java:42)
	at org.elixir_lang.annotator.Callable.annotate(Callable.kt:39)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.runAnnotators(DefaultHighlightVisitor.java:121)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.visit(DefaultHighlightVisitor.java:86)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.runVisitors(GeneralHighlightingPass.java:351)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$collectHighlights$5(GeneralHighlightingPass.java:284)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:311)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$analyzeByVisitors$6(GeneralHighlightingPass.java:314)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:70)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:314)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectHighlights(GeneralHighlightingPass.java:281)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:225)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:84)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:55)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$null$1(PassExecutorService.java:429)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1106)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:422)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:421)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:397)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:164)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:204)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:395)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:161)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -215,6 +215,8 @@
 ### Bug Fixes
 * [#1762](https://github.com/KronicDeth/intellij-elixir/pull/1762) - [@KronicDeth](https://github.com/KronicDeth)
   * Support diffs in pattern matching added in Elixir `1.10.0` in the ExUnit formatter.  Ports [elixir-lang/elixir@98c6bba436cc4833363295e5fedd3f819504d79d](https://github.com/elixir-lang/elixir/commit/98c6bba436cc4833363295e5fedd3f819504d79d).
+* [#1773](https://github.com/KronicDeth/intellij-elixir/pull/1773) - [@KronicDeth](https://github.com/KronicDeth)
+  * Only require at least the number of children to safely get a specific child instead of exact number to allow getting prefix children when there are grammatical errors.
 
 ## v11.6.0
 ### Enhancements

--- a/gradle.properties
+++ b/gradle.properties
@@ -2,7 +2,7 @@
 # https://www.jetbrains.com/intellij-repository/releases
 # https://www.jetbrains.com/intellij-repository/snapshots
 
-baseVersion = 11.6.0
+baseVersion = 11.6.1
 ideaVersion = 2020.1
 javaVersion = 1.8
 javaTargetVersion = 1.8

--- a/resources/META-INF/changelog.html
+++ b/resources/META-INF/changelog.html
@@ -11,6 +11,10 @@
           elixir-lang/elixir@<tt>98c6bba</tt>
         </a>.
       </li>
+      <li>
+        Only require at least the number of children to safely get a specific child instead of exact number to allow
+        getting prefix children when there are grammatical errors.
+      </li>
     </ul>
   </li>
 </ul>

--- a/src/org/elixir_lang/psi/impl/QuotableKeywordPairImpl.kt
+++ b/src/org/elixir_lang/psi/impl/QuotableKeywordPairImpl.kt
@@ -33,7 +33,7 @@ object QuotableKeywordPairImpl {
     fun getKeywordValue(keywordPair: ElixirKeywordPair): Quotable {
         val children = keywordPair.children
 
-        assert(children.size == 2)
+        assert(children.size >= 2)
 
         return children[1] as Quotable
     }

--- a/src/org/elixir_lang/psi/impl/QuotableKeywordPairImpl.kt
+++ b/src/org/elixir_lang/psi/impl/QuotableKeywordPairImpl.kt
@@ -42,7 +42,7 @@ object QuotableKeywordPairImpl {
     fun getKeywordValue(noParenthesesKeywordPair: ElixirNoParenthesesKeywordPair): Quotable {
         val children = noParenthesesKeywordPair.children
 
-        assert(children.size == 2)
+        assert(children.size >= 2)
 
         return children[1] as Quotable
     }

--- a/src/org/elixir_lang/reference/module/UnaliasedName.kt
+++ b/src/org/elixir_lang/reference/module/UnaliasedName.kt
@@ -32,7 +32,7 @@ object UnaliasedName {
                 is ElixirAccessExpression -> {
                     val children = element.children
 
-                    assert(children.size == 1)
+                    assert(children.size >= 1)
 
                     down(children[0])
                 }

